api/v1alpha1/componentdetectionquery_types.go

```diff
@@ -35,6 +35,12 @@ type ComponentDetectionQuerySpec struct {
 	// Optional.
 	// +optional
 	Secret string `json:"secret,omitempty"`
+
+	// If should generate the component name instead of a predicted default value
```

Comment:
Do you mean "It should generate"?

Reply:
I meant "if". as it's a flag and defines if the component name should generate random chars

Reply:
updated the description to make it more clear